astrocut/cutout_processing.py

```diff
@@ -0,0 +1,429 @@
+    if len(headers) == 1:
+        return headers[0]
+    
+    uniform_cards = []
+    varying_keywords = []
+    n_vk = 0
+    
+    for kwd in headers[0]:
+        
+        # Skip checksums etc
+        if kwd in ('S_REGION', 'CHECKSUM', 'DATASUM'):
+            continue
+        
+        if (np.array([x[kwd] for x in headers[1:]]) == headers[0][kwd]).all():
+            uniform_cards.append(headers[0].cards[kwd])
+        else:
+            if constant_only:  # Removing non-constant kewords in this case
+                continue
+                
+            n_vk += 1
+            for i, hdr in enumerate(headers):
+                varying_keywords.append((f"F{i+1:02}_K{n_vk:02}", kwd, "Keyword"))
+                varying_keywords.append((f"F{i+1:02}_V{n_vk:02}", hdr[kwd], "Value"))
+                varying_keywords.append((f"F{i+1:02}_C{n_vk:02}", hdr.comments[kwd], "Comment"))
+
+                
+    return fits.Header(uniform_cards+varying_keywords)
+
+
+def _get_bounds(x, y, size):
+    """
+    Given an x,y coordinates (single or lists) and size, return the bounds of the
+    described area(s) as [[[x_min, x_max],[y_min, y_max]],...].
+    """
+    x = np.array(np.atleast_1d(x))
+    y = np.array(np.atleast_1d(y))
+
+    lower_x = np.rint(x - size[0]/2)
+    lower_y = np.rint(y - size[1]/2)
+   
+    return np.stack((np.stack((lower_x, lower_x + size[0]), axis=1),
+                     np.stack((lower_y, lower_y + size[1]), axis=1)), axis=1).astype(int)
+
+
+def _combine_bounds(bounds1, bounds2):
+    """
+    Given two bounds of the form [[x_min, x_max],[y_min, y_max]],
+    combine them into a new [[x_min, x_max],[y_min, y_max]], that
+    encompasses both initial bounds.
+    """
+    
+    bounds_comb = np.zeros((2, 2), dtype=int)
+    bounds_comb[0, 0] = bounds1[0, 0] if (bounds1[0, 0] < bounds2[0, 0]) else bounds2[0, 0]
+    bounds_comb[1, 0] = bounds1[1, 0] if (bounds1[1, 0] < bounds2[1, 0]) else bounds2[1, 0]
+    bounds_comb[0, 1] = bounds1[0, 1] if (bounds1[0, 1] > bounds2[0, 1]) else bounds2[0, 1]
+    bounds_comb[1, 1] = bounds1[1, 1] if (bounds1[1, 1] > bounds2[1, 1]) else bounds2[1, 1]
+    
+    return bounds_comb
+
+
+def _area(bounds):
+    """
+    Given bounds of the form [[x_min, x_max],[y_min, y_max]] return
+    the area of the described rectangle.
+    """
+    return (bounds[0, 1] - bounds[0, 0]) * (bounds[1, 1] - bounds[1, 0])
+
+
+def _get_args(bounds, img_wcs):
+    """
+    Given bounds of the form [[x_min, x_max],[y_min, y_max]] and a
+    `~astropy.wcs.WCS` object return a center coordinate and size
+    ([ny, nx] pixels) suitable for creating a rectangular cutout.
+    """
+    nx = bounds[0, 1]-bounds[0, 0]
+    ny = bounds[1, 1]-bounds[1, 0]
+    x = nx/2 + bounds[0, 0]
+    y = ny/2 + bounds[1, 0]
+    return {"coordinates": img_wcs.pixel_to_world(x, y),
+            "size": (ny, nx)}
+
+
+# TODO: Put this in utils 
+def path_to_footprints(path, size, img_wcs, max_pixels=10000):
```

Comment:
I did not see this function used anywhere except in the tests. I guess that is used elsewhere and that's why you have the todo to move it to utils?

Reply:
Yes, sorry, I should have been more clean in the PR comment. See my edited comment about the moving target tesscut workflow.

Reply:
TLDR: Moving target tesscut uses this function and `center_on_path` for the full workflow, but they are not dependent on each other individually.